### ISSUE TITLE
Calling df.loc with multiple arguments results in KeyError

### ISSUE THREAD

**Issue by naren-ponder** [modin-project/modin#4354]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: macOS Version 11.6.4
- **Modin version** (`modin.__version__`): 0.14.0
- **Python version**: Python 3.8.11
- **Code we can use to reproduce**:

```
import modin.pandas as pd
import numpy as np

arrays = [
    np.array(["bar", "bar", "baz", "baz", "foo", "foo", "qux", "qux"]),
    np.array(["one", "two", "one", "two", "one", "two", "one", "two"]),
]
df = pd.DataFrame(np.random.randn(8, 4), index=arrays)
df.loc['bar', 'one']
```
Resulting Error:
```
KeyError                                  Traceback (most recent call last)
<ipython-input-7-5557f8ed36a3> in <module>
----> 1 df.loc['bar', 'one']

~/Desktop/modin/modin/pandas/indexing.py in __getitem__(self, key)
    636             return self._handle_boolean_masking(row_loc, col_loc)
    637 
--> 638         row_lookup, col_lookup = self._compute_lookup(row_loc, col_loc)
    639         result = super(_LocIndexer, self).__getitem__(row_lookup, col_lookup, ndim)
    640         if isinstance(result, Series):

~/Desktop/modin/modin/pandas/indexing.py in _compute_lookup(self, row_loc, col_loc)
    843                         else axis_loc
    844                     )
--> 845                     raise KeyError(missing_labels)
    846 
    847             if isinstance(axis_lookup, pandas.Index) and not is_range_like(axis_lookup):

KeyError: array(['one'], dtype='<U3')
```

Expected Output (with pandas):
```
0    0.395674
1   -0.426304
2    0.273483
3   -0.702982
Name: (bar, one), dtype: float64
```

### Describe the problem

Calling df.loc with multiple arguments results in Modin believing there are missing labels and therefore raises a KeyError.

**Comment by anmyachev** [comment]
@naren-ponder do you find the behavior strange? It would be more expected if it would be necessary to explicitly pass the tuple to work with the multi-index, like `df.loc[(bar, one)]`. 

If this behavior is wrong in pandas itself, maybe we should not repeat it?

**Comment by anmyachev** [comment]
The behavior of pandas in this case is not erroneous, I looked at the [docs](https://pandas.pydata.org/docs/user_guide/advanced.html#basic-indexing-on-axis-with-multiindex). So we definitely need to fix the case.

However, the previous question is still relevant.

**Comment by alvin-chang** [comment]
I got the same error, thus upvoting this issue.

**Comment by YarShev** [comment]
@anmyachev, if Modin behavior does not match the pandas behavior, we issue a warning like this.
https://github.com/modin-project/modin/blob/f41432c1c746c6a6186c376594c6c7f7dd24cdb5/modin/core/storage_formats/pandas/query_compiler.py#L2038

**Comment by naren-ponder** [comment]
@alvin-chang An easy workaround for this issue would be to separate out the calls to `.loc`. For instance in the case listed above you could do `df.loc['bar'].loc['one']`. This should unblock you while we work towards putting in a fix.

**Comment by RehanSD** [comment]
This also fails the case where we try and `.loc` by a row and col at the same time:
```python
import modin.pandas as pd
import numpy as np

arrays = [
    np.array(["bar", "bar", "baz", "baz", "foo", "foo", "qux", "qux"]),
    np.array(["one", "two", "one", "two", "one", "two", "one", "two"]),
]
df = pd.DataFrame(np.random.randn(8, 4), index=arrays)
df.loc['bar', 0]
```
gives us an error that it can't find key `0`, but on pandas, we get:
```python
one   -0.681044
two    1.607220
Name: 0, dtype: float64
```

**Comment by pyrito** [comment]
This issue seems to be resolved as of version `0.15.0+75.g3f985ed6` (latest master as of Aug. 11), so I'm going to go ahead and close this issue if that's ok with folks.

**Comment by mvashishtha** [comment]
This works at version 80c7891de7b6754a08d886895a110c2512c88e89.
